builtin/providers/aws/resource_aws_opsworks_stack.go

```diff
@@ -0,0 +1,463 @@
+			m["url"] = *v.URL
+		}
+		if v.Username != nil {
+			m["username"] = *v.Username
+		}
+		if v.Password != nil {
+			m["password"] = *v.Password
+		}
+		if v.Revision != nil {
+			m["revision"] = *v.Revision
+		}
+		if v.SSHKey != nil {
+			m["ssh_key"] = *v.SSHKey
+		}
+		nv = append(nv, m)
+	}
+
+	err := d.Set("custom_cookbooks_source", nv)
+	if err != nil {
+		// should never happen
+		panic(err)
+	}
+}
+
+func resourceAwsOpsworksStackRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AWSClient).opsworksconn
+
+	req := &opsworks.DescribeStacksInput{
+		StackIDs: []*string{
+			aws.String(d.Id()),
+		},
+	}
+
+	log.Printf("[DEBUG] Reading OpsWorks stack: %s", d.Id())
+
+	resp, err := client.DescribeStacks(req)
+	if err != nil {
+		if awserr, ok := err.(awserr.Error); ok {
+			if awserr.Code() == "ResourceNotFound" {
+				d.SetId("")
+				return nil
+			}
+		}
+		return err
+	}
+
+	stack := resp.Stacks[0]
+	d.Set("name", stack.Name)
+	d.Set("region", stack.Region)
+	d.Set("default_instance_profile_arn", stack.DefaultInstanceProfileARN)
+	d.Set("service_role_arn", stack.ServiceRoleARN)
+	d.Set("default_availability_zone", stack.DefaultAvailabilityZone)
+	d.Set("default_os", stack.DefaultOs)
+	d.Set("default_root_device_type", stack.DefaultRootDeviceType)
+	d.Set("default_ssh_key_name", stack.DefaultSSHKeyName)
+	d.Set("default_subnet_id", stack.DefaultSubnetID)
+	d.Set("hostname_theme", stack.HostnameTheme)
+	d.Set("use_custom_cookbooks", stack.UseCustomCookbooks)
+	d.Set("use_opsworks_security_groups", stack.UseOpsWorksSecurityGroups)
+	d.Set("vpc_id", stack.VPCID)
+	if stack.Attributes != nil {
+		if color, ok := (*stack.Attributes)["Color"]; ok {
+			d.Set("color", color)
+		}
+	}
+	if stack.ConfigurationManager != nil {
+		d.Set("configuration_manager_name", stack.ConfigurationManager.Name)
+		d.Set("configuration_manager_version", stack.ConfigurationManager.Version)
+	}
+	if stack.ChefConfiguration != nil {
+		d.Set("berkshelf_version", stack.ChefConfiguration.BerkshelfVersion)
+		d.Set("manage_berkshelf", stack.ChefConfiguration.ManageBerkshelf)
+	}
+	resourceAwsOpsworksSetStackCustomCookbooksSource(d, stack.CustomCookbooksSource)
+
+	return nil
+}
+
+func resourceAwsOpsworksStackCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AWSClient).opsworksconn
+
+	err := resourceAwsOpsworksStackValidate(d)
+	if err != nil {
+		return err
+	}
+
+	req := &opsworks.CreateStackInput{
+		DefaultInstanceProfileARN: aws.String(d.Get("default_instance_profile_arn").(string)),
+		Name:                    aws.String(d.Get("name").(string)),
+		Region:                  aws.String(d.Get("region").(string)),
+		ServiceRoleARN:          aws.String(d.Get("service_role_arn").(string)),
+		VPCID:                   aws.String(d.Get("vpc_id").(string)),
+		DefaultSubnetID:         aws.String(d.Get("default_subnet_id").(string)),
+		DefaultAvailabilityZone: aws.String(d.Get("default_availability_zone").(string)),
```

Comment:
`VPCID`, `DefaultSubnetID` and `DefaultAvailabilityZone` should not be filled here because we set these values below.

After fixing this, acc tests for stack will be all green.